app/profile/profile-service.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * Created by vytautassugintas on 09/04/16.
+ */
+"use strict";
+angular.module("myApp")
+    .factory("ProfileService", ProfileService);
+
+ProfileService.$inject = [
+    "User",
+    "Kudos",
+    "Auth",
+    "$q",
+    "$rootScope"
+];
+
+function ProfileService(userBackend, kudosBackend, authBackend, q, $rootScope) {
+    var service = {
+        userHome : UserHome,
+        updateUser: UpdateUser,
+        remainingKudos: RemainingKudos,
+        incomingKudos: IncomingKudos,
+        receivedKudos: ReceivedKudos,
+        send: SendKudos,
+        checkUser : CheckUser,
+        logout: Logout
+    };
+    return service;
+
+    function UserHome(){
+        var deferred = q.defer();
+        userBackend.home().then(function (response) {
```

Comment:
In most of service methods you are additionally using $q. It's redundant in many cases because you are not modifying promises. Simply return userBackend.home()